### ISSUE TITLE
chips(autocomplete): when no match required, 2 chips created on 'enter' key

### ISSUE THREAD

**Issue by apalayret** [angular/material#10640]
**Actual Behavior**:
 -  `What is the issue? *` When selecting an input from autocomplete (with no match required) with keyboard on IE, we end up with 2 chips, the one selected, and a new one matching the `search-text`
 -  `What is the expected behavior?` Same one as mouse selection, only end up with 1 chip (the one selected)

**CodePen** (or steps to reproduce the issue): *
 -  `CodePen Demo which shows your issue:` https://codepen.io/anon/pen/RVZWeX or https://material.angularjs.org/latest/demo/chips ("Custom Inputs" => "Use md-autocomplete to build an ordered set of chips.", no match required)
 -  `Details:` On IE, enter the begining of an existing input (as `bro`), and select **with the keyboard** one of the inputs on the dropdown.
If you select it with the mouse, it will work as expected, but with the keyboard, you will end up with 2 more chips, the one selected but also a new one
![image](https://cloud.githubusercontent.com/assets/1026843/25662352/821fe3e8-3014-11e7-9998-fc2620e7c53f.png)

**AngularJS Versions**: *
 -  `AngularJS Version:` 1.5.11
 -  `AngularJS Material Version:` 1.1.3

**Additional Information**:
 -  `Browser Type: *` IE
 -  `Browser Version: *` 11 at least
 -  `OS: *` Windows 7.1
 -  `Stack Traces:` N/A

**Comment by RudolfFrederiksen** [comment]
When combining `chips` with `autocomplete`, the chipsController uses the following code to add a listener on autocomplete generated input:
```
  // Bind to keydown and focus events of input
  inputElement
      .attr({ tabindex: 0 })
      .on('keydown', function(event) { scopeApplyFn(event, ctrl.inputKeydown) })
 ...
```

When a keyboard event is triggered on the autocomplete input, the following code is executed first from autocompleteController:
```
  function keydown (event) {
    switch (event.keyCode) {
   ...
      case $mdConstant.KEY_CODE.ENTER:
        if (ctrl.hidden || ctrl.loading || ctrl.index < 0 || ctrl.matches.length < 1) return;
        if (hasSelection()) return;
        event.stopPropagation();
        event.preventDefault();
        select(ctrl.index);
        break;
  ...
```

AutocompleteController stops propagation and prevents default behavior.
Then, even if the event won't go on the parent element, other registered listeners on this input will continue to be called.

Thus chipsController.inputKeydown() is called. (registration has been made on the first code sample)
```
MdChipsCtrl.prototype.inputKeydown = function(event) {
  var chipBuffer = this.getChipBuffer();
  // If we have an autocomplete, and it handled the event, we have nothing to do
  if (this.autocompleteCtrl && event.isDefaultPrevented && event.isDefaultPrevented()) {
    return;
  }
 ...
```

The problem is that on IE11 on this particular setup. The keyboard event passed to this function, doesn't have its preventDefault attribute set to `false`.
So some reasons, it is still set to `false` while autocompleteController explicitly set it to `true`.

I havent found out yet the exact reason of this behavior. And I guess using `stopImmediatePropagation()` might have some border effect on other components.
This shall to confirmed.

However I tested a simple event chains on IE11 with the following code:
https://jsfiddle.net/x9qkrjLb/

And during the call chains, the keyboard event has the expected behavior.
The first listener uses preventDefault.
The second listener has preventDefault set to true.

**Comment by RudolfFrederiksen** [comment]
By pushing the investigation a bit further

```
 // Bind to keydown and focus events of input
  inputElement
      .attr({ tabindex: 0 })
      .on('keydown', function(event) { scopeApplyFn(event, ctrl.inputKeydown) })
 ...
```

```
  var scopeApplyFn = function(event, fn) {
    scope.$evalAsync(angular.bind(ctrl, fn, event));
  };
```

Logging `event` before $evalAsync gave the expected result on IE11.
Any log on `event` after $evalAsync call returns an "unpreventedDefault" keyboard event.
This is really weird to me and I don't understand why this happen even after looking at angular base code.
https://github.com/angular/angular.js/blob/v1.5.11/src/ng/rootScope.js#L1010
https://github.com/angular/angular.js/blob/master/src/ng/parse.js#L1632
